packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -178,6 +187,129 @@ class SegmentedButton<T> extends StatefulWidget {
+  /// standard opacity adjustments for the pressed, focused, and
+  /// hovered states, one could write:
+  ///
+  /// ```dart
+  /// SegmentedButton<int>(
+  ///   style: SegmentedButton.styleFrom(
+  ///     foregroundColor: Colors.black,
+  ///     selectedForegroundColor: Colors.white,
+  ///     backgroundColor: Colors.amber,
+  ///     selectedBackgroundColor: Colors.red,
+  ///   ),
+  ///   segments: const <ButtonSegment<int>>[
+  ///     ButtonSegment<int>(
+  ///       value: 0,
+  ///       label: Text('0'),
+  ///       icon: Icon(Icons.calendar_view_day),
+  ///     ),
+  ///     ButtonSegment<int>(
+  ///       value: 1,
+  ///       label: Text('1'),
+  ///       icon: Icon(Icons.calendar_view_week),
+  ///     ),
+  ///     ButtonSegment<int>(
+  ///       value: 2,
+  ///       label: Text('2'),
+  ///       icon: Icon(Icons.calendar_view_month),
+  ///     ),
+  ///     ButtonSegment<int>(
+  ///       value: 3,
+  ///       label: Text('3'),
+  ///       icon: Icon(Icons.calendar_today),
+  ///     ),
+  ///   ],
+  ///   selected: const <int>{0},
+  ///   onSelectionChanged: (Set<int> selection) {},
+  /// ),
+  /// ```
+  static ButtonStyle styleFrom({
+    Color? foregroundColor,
+    Color? backgroundColor,
+    Color? selectedForegroundColor,
+    Color? selectedBackgroundColor,
+    Color? disabledForegroundColor,
+    Color? disabledBackgroundColor,
+    Color? shadowColor,
+    Color? surfaceTintColor,
+    double? elevation,
+    TextStyle? textStyle,
+    EdgeInsetsGeometry? padding,
+    Size? minimumSize,
+    Size? fixedSize,
+    Size? maximumSize,
+    BorderSide? side,
+    OutlinedBorder? shape,
+    MouseCursor? enabledMouseCursor,
+    MouseCursor? disabledMouseCursor,
+    VisualDensity? visualDensity,
+    MaterialTapTargetSize? tapTargetSize,
+    Duration? animationDuration,
+    bool? enableFeedback,
+    AlignmentGeometry? alignment,
+    InteractiveInkFeatureFactory? splashFactory,
+  }) {
+    final MaterialStateProperty<Color?>? foregroundColorProp =
+      (foregroundColor == null && disabledForegroundColor == null && selectedForegroundColor == null)
+        ? null
+        : _SegmentButtonDefaultColor(foregroundColor, disabledForegroundColor, selectedForegroundColor);
+    final MaterialStateProperty<Color?>? backgroundColorProp =
+      (backgroundColor == null && disabledBackgroundColor == null && selectedBackgroundColor == null)
+        ? null
+        : _SegmentButtonDefaultColor(backgroundColor, disabledBackgroundColor, selectedBackgroundColor);
+    final MaterialStateProperty<Color?>? overlayColor = (foregroundColor == null)
+      ? null
+      : _SegmentedButtonDefaultOverlay(foregroundColor);
+    final MaterialStateProperty<MouseCursor?>? mouseCursor = (enabledMouseCursor == null && disabledMouseCursor == null)
+      ? null
+      : _SegmentedButtonDefaultMouseCursor(enabledMouseCursor, disabledMouseCursor);
+    return ButtonStyle(
```

Comment:
Could this have been written in terms of ButtonStyle.styleFrom()? If there are any differences you could write something like `return ButtonStyle.styleFrom(...).copyWith(...)`

Reply:
@AcarFurkan seems to be lost so I've addressed this feedback.

Added `TextButton.styleFrom` with `.copyWith` for background, foreground, and overlay color.